virtool/references/db.py

```diff
@@ -433,6 +434,70 @@ async def update_reference(self):
         })
 
 
+class CreateIndexJSONTask(virtool.tasks.task.Task):
+
+    def __init__(self, app, task_id):
+        super().__init__(app, task_id)
+
+        self.steps = [
+            self.check_index_json_files,
+            self.create_index_json_files
+        ]
+
+        self.index_without_json = []
+
+    async def check_index_json_files(self):
```

Comment:
Drop this method and move to `create_index_json_files`. You can just use a Mongo query to replace lines 452-454:
```python
db.indexes.find({"has_json": {"$ne": True}})
```

Reply:
For reference: https://docs.mongodb.com/manual/reference/operator/query/